### PR TITLE
Support for repos from multiple architectures

### PR DESCRIPTION
**Pull request by vuntz**
We always deal with x86_64 attributes, but do not allow setting the
repos for non-x86_64 architectures.

~~Note that this contains https://github.com/yast/yast-crowbar/pull/33~~

cc @jsuchome

### CODE REVIEW

**Comment by jsuchome** [comment]
Tested: writing new format works, and also reading from it.

We don't want to support reading from previous format of provisioner.json, right?

Also, could you use some constant instead of that "x86_64" string everywhere?

**Comment by vuntz** [comment]
> We don't want to support reading from previous format of provisioner.json, right?

I would think it's not worth it.

> Also, could you use some constant instead of that "x86_64" string everywhere?

Do you prefer a global constant, or one per file?

**Comment by jsuchome** [comment]
I think global is good

**Comment by vuntz** [comment]
@jsuchome done

**Comment by jsuchome** [comment]
+1

**Comment by lslezak** [comment]
Um, is the `@default_arch` actually set somewhere? For me it seems it's always set to `x86_64` or did I overlook something?

Otherwise LGTM.

**Comment by mvidner** [comment]
- This looks like an adaptation to a bigger change. Is there a Bugzilla item/GitHub issue  to be mentioned in .changes?
- Is the (new) format of provisioner.json documented somewhere? Maybe you want to put a hyperlink near the code for easy reference.
- Some lines have overflown the right margin, maybe you want to break them up and name the intermediate values for clarity.

Anyway, I don't feel like applying YaST team review process to this, so consider this a drive-by review, 'k thanks bye!

**Comment by jsuchome** [comment]
@lslezak Yes, it's x86_64 only, but still better than using the value everywhere, I think

**Comment by lslezak** [comment]
I see, so it's a clean up or preparation for future changes... Refactoring to a single place is always a good idea so LGTM.

**Comment by vuntz** [comment]
For the record, the rebase of this on top of https://github.com/yast/yast-crowbar/issues/35 is at https://github.com/vuntz/yast-crowbar/compare/yaml-repos...vuntz:x86_64-arch-yaml (and I actually made it support other architectures than just x86_64).

**Comment by vuntz** [comment]
Rebased, but not tested yet.

**Comment by vuntz** [comment]
Seems to work now.

File: src/include/crowbar/complex.rb [hunk 1]
```diff
@@ -469,20 +472,22 @@ def InitRepoURL(id)
...
+                repo["url"] || "",
+                (repo["ask_on_error"] || false) ?  UI.Glyph(:CheckMark) : " ",
+                "#{@platform2label[platform] || platform} (#{arch})"
+              )
+          end
         end
       end
       UI.ChangeWidget(Id(id), :Items, repo_items)
-      UI.ChangeWidget(Id(id), :CurrentItem, "#{@current_repo_platform}|#{@current_repo}") unless @current_repo.empty?
+      UI.ChangeWidget(Id(id), :CurrentItem, "#{@current_repo_platform}|#{@current_repo}|#{@current_repo_id}") unless @current_repo.empty?
```
Comment (jsuchome) [inline src/include/crowbar/complex.rb]: Not sure where is the problem, but @current_repo_id (in some cases?) seem to be empty

Comment (jsuchome) [inline src/include/crowbar/complex.rb]: Oh, it's always empty, it's a new variable! :-)

Comment (vuntz) [inline src/include/crowbar/complex.rb]: Wow. How come this works for me with such a stupid bug?

Comment (vuntz) [inline src/include/crowbar/complex.rb]: fixed

Comment (jsuchome) [inline src/include/crowbar/complex.rb]: It works well, that's just minor UI glitch, probably only relevant in Qt anyway.

**Comment by jsuchome** [comment]
+ after the rebase

**Pull request merged**